api/v1alpha1/nonadminbackup_types.go

```diff
@@ -21,16 +21,37 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// NonAdminBackupPhase is a simple one high-level summary of the lifecycle of an NonAdminBackup.
+// +kubebuilder:validation:Enum=New;BackingOff;Created
+type NonAdminBackupPhase string
+
+const (
+	// NonAdminBackupPhaseNew - NonAdminBackup resource was accepted by the OpenShift cluster, but it has not yet been processed by the NonAdminController
+	NonAdminBackupPhaseNew NonAdminBackupPhase = "New"
+	// NonAdminBackupPhaseBackingOff - Velero Backup object was not created due to NonAdminBackup error (configuration or similar)
+	NonAdminBackupPhaseBackingOff NonAdminBackupPhase = "BackingOff"
+	// NonAdminBackupPhaseCreated - Velero Backup was created. The Phase will not have additional informations about the Backup.
+	NonAdminBackupPhaseCreated NonAdminBackupPhase = "Created"
+)
+
+// NonAdminCondition are used for more detailed information supporing NonAdminBackupPhase state.
+// +kubebuilder:validation:Enum=BackupAccepted;BackupQueued
+type NonAdminCondition string
```

Comment:
I do not like declaring things on a file that are not used on that file. Would move to file that uses it

kubebuilder annotation here does not do anything, right?

Reply:
I have no idea what is the comment here about, could you give me more information?

Reply:
check CRD yaml, it does not mention NonAdminCondition, right? then `// +kubebuilder:validation:Enum=BackupAccepted;BackupQueued` does not do anything

This type is declared here, but never used here, only by other files. If it is used by more then one file, would move to constants file, otherwise, just to the file that uses this type

Reply:
this is still never used in the file it was declared, right :question:

Reply:
It is used in this file, few lines below as a NonAdminCondition:
```
const (
	NonAdminConditionAccepted NonAdminCondition = "Accepted"
	NonAdminConditionQueued   NonAdminCondition = "Queued"
)
```

Reply:
the `NonAdminCondition` type is not used in this file (it is just created and its allowed values declared), it should be moved to another file

Reply:
it is used by functions and nab controller, so common folder

we can add it to constants directly, or create a types file in their first

Reply:
here you go